code/modules/reagents/chemistry/recipes/pyrotechnics.dm

```diff
@@ -0,0 +1,175 @@
+/datum/chemical_reaction/reagent_explosion
+	name = "Generic explosive"
+	id = "reagent_explosion"
+	var/strengthdiv = 10
+	var/modifier = 0
+
+/datum/chemical_reaction/reagent_explosion/on_reaction(datum/reagents/holder, created_volume)
+	var/turf/T = get_turf(holder.my_atom)
+	var/inside_msg
+	if(ismob(holder.my_atom))
+		var/mob/M = holder.my_atom
+		inside_msg = " inside [ADMIN_LOOKUPFLW(M)]"
+	var/lastkey = holder.my_atom.fingerprintslast
+	var/touch_msg = "N/A"
+	if(lastkey)
+		var/mob/toucher = get_mob_by_key(lastkey)
+		touch_msg = "[ADMIN_LOOKUPFLW(toucher)]"
+	message_admins("Reagent explosion reaction occurred at [ADMIN_VERBOSEJMP(T)][inside_msg]. Last Fingerprint: [touch_msg].")
+	log_game("Reagent explosion reaction occurred at [AREACOORD(T)]. Last Fingerprint: [lastkey ? lastkey : "N/A"]." )
+	var/datum/effect_system/reagents_explosion/e = new()
+	e.set_up(modifier + round(created_volume/strengthdiv, 1), T, 0, 0)
+	e.start()
+	holder.clear_reagents()
+
+
+/datum/chemical_reaction/reagent_explosion/nitroglycerin
+	name = "Nitroglycerin"
+	id = "nitroglycerin"
+	results = list("nitroglycerin" = 2)
+	required_reagents = list("glycerol" = 1, "facid" = 1, "sacid" = 1)
+	strengthdiv = 2
+
+/datum/chemical_reaction/reagent_explosion/nitroglycerin/on_reaction(datum/reagents/holder, created_volume)
+	if(holder.has_reagent("stabilizing_agent"))
```

Comment:
Stabilizing Agent is not present yet on our reagents module, but you should be able to port it here without problem.
Guess I'll have to update the interaction between heating tools and beakers for the temperature stuff to work now that recipes with temperature requirement are going in

Reply:
This still needs to be answered.

Reply:
This is not a question, really. Or was there a suggestion implied?